src/bsp/src/stm32/src/HostUart.cpp

```diff
@@ -62,11 +70,22 @@ bool HostUart::send(const uint8_t* buffer, uint16_t length) {
     return ret;
 }
 
-int32_t HostUart::receive(uint8_t* buffer, uint16_t length) const {
-    (void)buffer;
-    (void)length;
-    // TODO: Implement
-    return 0;
+bool HostUart::receive(uint8_t* buffer, uint16_t length) {
+    if (buffer == NULL || length > m_streamMemory.size()) {
```

Comment:
isn't better to compare with nullptr? I believe NULL means 0

Reply:
nullptr is also equal to 0, but will change because we are in c++